src/main/java/si/mazi/rescu/clients/ApacheConnection.java

```diff
@@ -0,0 +1,225 @@
+            response = client.execute(request);
+            
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+        executed = true;
+    }
+
+    private static HttpRequestBase createRequest(HttpMethod method, String url) {
+        switch(method) {
+            case GET: return new HttpGet(url); 
+            case POST: return new HttpPost(url);
+            case PUT: return new HttpPut(url);
+            case DELETE: return new HttpDelete(url);
+            case HEAD: return new HttpHead(url); 
+            case OPTIONS: return new HttpOptions(url);
+            case PATCH: return new HttpPatch(url);
+            default: throw new RuntimeException("Not supported http method: " + method);
+        }
+    }
+
+    @Override
+    public String getHeaderField(String name) {
+        try {
+            return res.getFirstHeader(name).getValue();
+        } catch (Exception e) {
+            return null;
+        }
+    }
+    
+    @Override
+    public void setRequestMethod(HttpMethod method) throws ProtocolException {
+        this.method = method;
+    }
+
+    @Override
+    public String getRequestMethod() {
+        return method.name();
+    }
+
+    @Override
+    public OutputStream getOutputStream() throws IOException {
+        if(out == null) {
+            out = new ByteArrayOutputStream();
+        }
+        return out;
+    }
+    
+    @Override
+    public InputStream getInputStream() throws IOException {
+        return response.getEntity().getContent();
+    }
+    @Override
+    public InputStream getErrorStream() throws IOException {
+        return getInputStream();
+    }
+    
+    @Override
+    public void setDoOutput(boolean dooutput) {
+     // @TODO ???
+    }
+
+    @Override
+    public void setDoInput(boolean doinput) {
+     // @TODO ???
+    }
+    
+    @Override
+    public int getResponseCode() throws IOException {
+        exec();
+        return response.getStatusLine().getStatusCode();
+    }
+
+    @Override
+    public Map<String, List<String>> getHeaderFields() {
+        
+        Map<String, List<String>> result = new HashMap<>();
+        Stream.of(response.getAllHeaders()).forEach(h -> {
+            List<String> list = result.get(h.getName());
+            if (list == null) {
+                list = new ArrayList<>();
+                result.put(h.getName(), list);
+            }
+            list.add(h.getValue());
+        });
+        
+        return result;
+    }
+
+    @Override
+    public void addHeader(String key, String value) {
+        if ("Content-Length".equals(key)) { // need to drop this header here, otherwise we get "Content-Length header already present"
+            return;
+        }
+        headers.put(key, value);
+    }
+
+    @Override
+    /**
+     * @param timeout an {@code int} that specifies the timeout value to be used in milliseconds
+     */
+    public void setReadTimeout(int readTimeout) {
+     // @TODO ???
```

Comment:
Please check if this should use  requestConfig.setSocketTimeout(), or log a warning that this is ignored.